app/main/routes.py

```diff
@@ -58,20 +59,54 @@ def index():
 
         # Persist word and meaning to database.
         slang_word = session.get("slang_word", None)
-        word_type = session.get("model_type", None)
+        model_type = session.get("model_type", None)
+        meaning = meaning_form.meaning.data
         user_id = current_user.get_id()
 
         if not slang_word:
             flash("You haven't generated a slang word yet...")
             return redirect(url_for("main.index"))
 
-        word_db_model = Word(
-            word=slang_word,
-            meaning=meaning_form.meaning.data,
-            type=word_type,
+        # TODO: Refactor this database persistence functionality to a separate method.
```

Comment:
Agreed, but can see how this might be tricky doing it for two separate classes, so I don't mind it for right now